img_cmd_training_new_dataset.py

```diff
@@ -0,0 +1,280 @@
+
+compared_models = {
+    "resnet18": resnet18,
+    "resnet34": resnet34
+}
+
+DATASET_DIR.ls()
+
+def filename_to_class(filename: str) -> str:
+    angle = float(filename.split("_")[1].split(".")[0].replace("p", "."))
+    if angle > 0:
+        return "left"
+    elif angle < 0:
+        return "right"
+    else:
+        return "forward"
+
+class ImageWithCmdDataset(Dataset):
+    def __init__(self, filenames):
+        """
+        Creates objects for class labels, class indices, and filenames.
+        
+        :param filenames: (list) a list of filenames that make up the dataset
+        """
+        self.class_labels = ['left', 'forward', 'right']
+        self.class_indices = {lbl:i for i, lbl in enumerate(self.class_labels)} # {'left': 0, 'forward': 1, 'right': 2}        
+        self.all_filenames = filenames
+        
+    def __len__(self):
+        """
+        Gives length of dataset.
+        
+        :return: (int) the number of filenames in the dataset
+        """
+        return len(self.all_filenames)
+
+    def __getitem__(self, index):
+        """
+        Gets the filename associated with the given index, opens the image at
+        that index, then uses the image's filename to get information associated
+        with the image such as its label and the label of the previous image.
+        
+        :param index: (int) number that represents the location of the desired data
+        :return: (tuple) tuple of all the information associated with the desired data
+        """
+        # The filename of the image given a specific index
+        img_filename = self.all_filenames[index]            
+        
+        # Opens image file and ensures dimension of channels included
+        img = Image.open(img_filename).convert('RGB')
+        # Resizes the image
+        img = img.resize((224, 224))
+        # Converts the image to tensor and 
+        img = torch.Tensor(np.array(img)/255)
+        # changes the order of the dimensions
+        img = img.permute(2,0,1)
+        
+        # Getting the current image's label
+        label_name = filename_to_class(str(img_filename))
+        label = self.class_indices[label_name]
+        
+        # Getting the previous image's label
+        # The default is 'forward'
+        cmd_name = 'forward'
+        
+        # If the index is not 0, the cmd is determined by the previous img_filename
+        if index != 0:
+            prev_img_filename = self.all_filenames[index-1]
+            cmd_name = filename_to_class(str(prev_img_filename))            
+        cmd = self.class_indices[cmd_name]
+        
+        # Data and the label associated with that data
+        return (img, cmd), label
+
+class cmd_model(nn.Module):
+    def __init__(self, arch: str, pretrained: bool):
+        super(cmd_model, self).__init__()
+        self.cnn = arch(pretrained=pretrained)
+        
+        self.fc1 = nn.Linear(self.cnn.fc.out_features + 1, 512)
```

Comment:
We probably want to make this part of the network less arbitrary. Make a note that we should try other architectures here.